java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -2023,4 +2027,27 @@ private Object convertByteArrayToGlideString(Object o) {
         }
         return o;
     }
+
+    @Override
+    public CompletableFuture<GlideString> dump(@NonNull GlideString key) {
+        GlideString[] arguments = new GlideString[] {key};
+        return commandManager.submitNewCommand(Dump, arguments, this::handleBytesOrNullResponse);
+    }
+
+    @Override
+    public CompletableFuture<String> restore(
+            @NonNull GlideString key, long ttl, @NonNull GlideString value) {
```

Comment:
I think commands should accept and return to user a `byte[]` for these commands - it is always a binary dump.
```
byte[] dump(GlideString key)
String restore(GlideString key, long ttl, byte[] data)
```